packages/mgt-chat/src/statefulClient/StatefulGraphChatClient.ts

```diff
@@ -855,17 +859,53 @@ detail: ${JSON.stringify(eventDetail)}`);
     if (this.emojiMatch(content)) {
       content = this.processEmojiContent(content);
     }
+    // Handle any mentions
+    const mentions = graphMessage?.mentions ?? [];
+    if (mentions) {
+      content = this.handleMentions(mentions, content);
+    }
 
     const imageMatch = this.graphImageMatch(content ?? '');
     if (imageMatch) {
       // if the message contains an image, we need to fetch the image and replace the placeholder
       result = this.processMessageContent(graphMessage, currentUser);
     } else {
-      result.currentValue = this.buildAcsMessage(graphMessage, currentUser, graphMessage.id, content);
+      result.currentValue = this.buildAcsMessage(graphMessage, currentUser, messageId, content);
     }
     return result;
   }
 
+  private handleMentions(mentions: ChatMessageMention[], content: string): string {
+    // Nit to remove unnecessary space between 2 mentions
+    content = content.replace(/&nbsp;<at/gim, '<at');
+    mentions.forEach((mention: ChatMessageMention) => {
+      const name = mention?.mentionText;
+      // Multiple entities can be mentioned: user, application, conversation etc
+      const user: NullableOption<Identity> | undefined = mention.mentioned?.user;
+      if (name && user) {
+        const { id } = user;
+        const mgtPerson = `<mgt-person
```

Comment:
This should probably make use of the `customElementHelper.prefix` to construct the tag, just in case someone is using this with disambiguation